contrib/upgrade.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+set -e
+
+# This is an update script for gitea installed via the binary distribution
+# from dl.gitea.io on linux as systemd service. It performs a backup and updates
+# Gitea in place.
+# NOTE: This adds the GPG Signing Key of the Gitea maintainers to the keyring.
+# Depends on: bash, curl, xz, sha256sum, gpg, which. optionally jq.
+# Usage:      [environment vars] upgrade.sh [version]
+#   See section below for available environment vars.
+#   When no version is specified, updates to the latest release.
+# Examples:
+#   upgrade.sh 1.15.10
+#   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
+
+# apply variables from environment
+: ${giteabin:="/usr/local/bin/gitea"}
+: ${giteahome:="/var/lib/gitea"}
+: ${giteaconf:="/etc/gitea/app.ini"}
+: ${giteauser:="git"}
+: ${sudocmd:="sudo"}
+: ${arch:="linux-amd64"}
+: ${backupopts:=""} # see `gitea dump --help` for available options
+
+function giteacmd {
+  "$sudocmd" --user "$giteauser" "$giteabin" --config "$giteaconf" --work-path "$giteahome" $@
+}
+
+function require {
+  for exe in $@; do
+    which $exe &>/dev/null || (echo "missing dependency '$exe'"; exit 1)
+  done
+}
+require curl xz sha256sum gpg
```

Comment:
systemctl should be added to this list as this script depends on a linux distribution using systemd as its service manager.
I have gitea running in multiple locations and none use systemd (gentoo, alpine-metal, alpine containers) but gitea is also managed via the package manager (be it v1.15)

Reply:
I think ` on linux as systemd service.` in the description of this script should be sufficient. If systemd is installed systemctl is installed as well. But it could be added as a safeguard, but people using things like init should have their own kind of way to do this, as we don't have a ready to go script for such configurations.